src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/AssemblyIdentityMustMatch.cs

```diff
@@ -75,7 +75,7 @@ private void RunOnAssemblySymbol(IAssemblySymbol left, IAssemblySymbol right, st
                     rightName,
                     rightIdentity));
             }
-            else if (Settings.StrictMode && !rightAssemblyPublicKeyToken.IsEmpty && !isRightRetargetable)
+            else if (Settings.StrictMode && !rightAssemblyPublicKeyToken.IsEmpty && !isRightRetargetable && !rightAssemblyPublicKeyToken.SequenceEqual(leftAssemblyPublicKeyToken))
```

Comment:
if the rightAssemblyPublicToken.IsEmpty and the left is not and the right assembly is not retargetable this would not add a difference, right?

Same for the if above with left pkt.

Reply:
right (empty and NonRetargetable left (non-empty and non-retargetable) will result in mismatch in normal mode only. we wont need to check for this condition.

Reply:
Right but if the public key token is empty and the assembly is not retargetable, then we would not error the difference, correct?

Reply:
yes if "base" dll token is empty then comparable dll can be null or not null and we wont error the difference.